File: tests/fixtures/multilang/sample.c
#include <stdio.h>
#include <stdlib.h>

/* A sample C file for parser testing */

struct Config {
    char* name;
    int value;
};

enum Status {
    ACTIVE,
    INACTIVE,
    PENDING
};

typedef struct Config ConfigAlias;

int validate_config(struct Config* config) {
    return config->name != NULL && config->value > 0;
}

void process_config(struct Config* config) {
    if (validate_config(config)) {
        printf("Valid: %s\n", config->name);
    }
}

File: tests/fixtures/multilang/sample.rs
/// A sample Rust module for parser testing
pub struct Config {
    pub name: String,
    pub value: i32,
}

pub trait Validator {
    fn validate(&self) -> bool;
}

impl Validator for Config {
    fn validate(&self) -> bool {
        !self.name.is_empty() && self.value > 0
    }
}

pub enum Status {
    Active,
    Inactive,
    Pending,
}

/// Processes a config and returns its status
pub fn process_config(config: &Config) -> Status {
    if config.validate() {
        Status::Active
    } else {
        Status::Inactive
    }
}

const MAX_RETRIES: u32 = 3;
